api/net/Areas/Services/Controllers/ContentReferenceController.cs

```diff
@@ -104,10 +104,13 @@ public IActionResult UpdateKafka(ContentReferenceModel model)
     {
         var reference = _service.FindByKey(model.Source, model.Uid);
         if (reference == null) return new NoContentResult();
-        reference.Offset = model.Offset;
-        reference.Partition = model.Partition;
-        var result = _service.UpdateAndSave(reference);
-        return new JsonResult(new ContentReferenceModel(result));
+        lock (reference)
```

Comment:
What are you hoping to achieve with this?  Each request to the API has it's own scope.  They won't ever lock.  Even if a request did share scope the reference would be old and result in the same error anyway.

Reply:
Currently it looks like we have much less errors than before, and the remaining a few errors seem to be related to the updates here, using lock here just to test and see if this could make any differences and provide kind of a peace of mind? and we could remove it later if we still have the issue? What do you think? Thanks

Reply:
Well this won't change anything as I described above.

I believe this endpoint is only ever called when content has been received `IngestAction.ContentReceivedAsync(...)`

This is essentially what is happening

a) Ingest Service sends request to UpdateKafka(...) method - completes successfully
b) Ingest Service sends request to UpdateKafka(...) method with old content reference - fails due to the first request

If it is the Syndication Service (which it probably is).  You need to find out why it has updated a `ContentReference` between the time it processed a `Content` object.

Reply:
Updated, thanks